packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -758,7 +759,9 @@ export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget
         super.dispose();
     }
 
-    protected resizeTerminal(): void {
+    protected resizeTerminal = debounce(() => this.doResizeTerminal(), 50);
+
+    protected doResizeTerminal(): void {
         const geo = this.fitAddon.proposeDimensions();
```

Comment:
During automated tests, this function call yields `undefined` for some reason. I'm not sure why, since it really shouldnt. However, we should just guard against it to fix the tests. See also [here](https://github.com/eclipse-theia/theia/actions/runs/4695903441/jobs/8330498515#step:9:1855)

Reply:
I believe we have a race condition. The previous test creates a terminal widget and it's then disposed it in the `afterEach()` before the following (failing) test is run. What I think happens is that between the moment the resize is triggered and the `debounce` delay has expired, the previous test is finished, and the terminal widget has been disposed. 

It's not something likely to happen at the human scale, but since we introduce a delay here, it seems reasonable to make sure that the widget is not disposed before trying to access its internals. e.g.:
```typescript
protected doResizeTerminal(): void {
   if (this.isDisposed) {
      return;
   }
   const geo = this.fitAddon.proposeDimensions();
[...]
```